### PR TITLE
Don't use UUIDs to store AIPs in the Wellcome Storage; use identifiers from metadata.csv

### PR DESCRIPTION
**Pull request by alexwlchan**
For https://github.com/wellcometrust/platform/issues/4084

### CODE REVIEW

File: storage_service/locations/models/wellcome.py [hunk 1]
```diff
@@ -87,6 +87,120 @@ def mkdir_p(dirpath):
...
+    # fall back to the UUID.
+    bag_dir = os.path.join(temp_dir, os.listdir(temp_dir)[0])
+    assert os.path.exists(bag_dir)
+    LOGGER.debug("Expanded bag into directory %s" % bag_dir)
+    bag = bagit.Bag(bag_dir)
+
+    manifest_csvs = [
+        name
+        for name in bag.payload_files()
+        if name.endswith("/metadata.csv")
```
Comment (alexwlchan) [inline storage_service/locations/models/wellcome.py]: This check will fail if there's a random file in the digital archive called `metadata.csv`.

We can say exactly which it is – in a path of a particular format and depth.

**Review by alexwlchan — COMMENTED** [review]
This is failing a bunch of tests right now (and will need new ones for the new code) – I wanted to get  sense check of whether this was sensible/desirable before pushing on.

**Comment by helenst** [comment]
Been testing out / stepping through - this seems good from an Archivematica perspective. (though will wait until tests are in place and we know this is going to be used before I do full code review)

Also means that updates to a bag can happen not just through a reingest but through a new transfer / ingest of the file, so long as the identifiers are the same. (with Archivematica UUIDs, it would be stored under different identifiers each time)

**Comment by alexwlchan** [comment]
@helenst Awesome, thanks for the review!

I'm going to park this briefly while I tinker with the Wellcome Storage so we can do slashes in the external identifier – but glad to hear the idea has legs. I’ll let you know when it’s ready for another review.

**Pull request merged**